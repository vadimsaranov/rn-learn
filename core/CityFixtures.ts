export const cityFixture = {
  coord: {
    lon: 25.2798,
    lat: 54.6892,
  },
  weather: [
    {
      id: 804,
      main: 'Clouds',
      description: 'overcast clouds',
      icon: '04d',
    },
  ],
  base: 'stations',
  main: {
    temp: 276.87,
    feels_like: 273.69,
    temp_min: 276.64,
    temp_max: 277.12,
    pressure: 1021,
    humidity: 80,
    sea_level: 1021,
    grnd_level: 1001,
  },
  visibility: 7000,
  wind: {
    speed: 3.6,
    deg: 160,
  },
  clouds: {
    all: 100,
  },
  dt: 1740735734,
  sys: {
    type: 1,
    id: 1883,
    country: 'LT',
    sunrise: 1740719461,
    sunset: 1740757930,
  },
  timezone: 7200,
  id: 593116,
  name: 'Vilnius',
  cod: 200,
};
