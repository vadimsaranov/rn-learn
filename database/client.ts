import { drizzle } from 'drizzle-orm/expo-sqlite/driver';
import { openDatabaseSync } from 'expo-sqlite';

const DATABASE_NAME = 'rn-learn.db';

export const expoDb = openDatabaseSync(DATABASE_NAME, { enableChangeListener: true });
export const appDatabase = drizzle(expoDb);
