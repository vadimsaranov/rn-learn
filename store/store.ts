import AsyncStorage from '@react-native-async-storage/async-storage';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { authReducer } from '@store/slices/authSlice';
import { biometricsReducer } from '@store/slices/biometricsSlice';
import { sessionReducer } from '@store/slices/sessionSlice';
import { userReducer } from '@store/slices/userSlice';
import { useDispatch, useSelector } from 'react-redux';
import { persistReducer, persistStore } from 'redux-persist';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  blacklist: ['session'],
};

const rootReducer = combineReducers({
  user: userReducer,
  auth: authReducer,
  session: sessionReducer,
  biometrics: biometricsReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;

export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();
