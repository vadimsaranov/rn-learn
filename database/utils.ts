import { appDatabase } from '@database/client';
import { cityTable } from '@database/schemas/cityTable';
import { userTable } from '@database/schemas/userTable';
import { weatherTable } from '@database/schemas/weatherTable';

export const wipeDatabase = async () => {
  await appDatabase.delete(cityTable);
  await appDatabase.delete(weatherTable);
  await appDatabase.delete(userTable);
};
