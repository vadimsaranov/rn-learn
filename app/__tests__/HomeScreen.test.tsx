import CitiesContextProvider from '@context/CitiesContext';
import { cityFixture } from '@core/CityFixtures';
import { appDatabase } from '@database/client';
import { wipeDatabase } from '@database/utils';
import { persistor, store } from '@store/store';
import { migrate } from 'drizzle-orm/expo-sqlite/migrator';
import { fireEvent, renderRouter, screen, waitFor } from 'expo-router/testing-library';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import HomeLayout from '../(tabs)/(home)/_layout';
import AddOrEditCityWeather from '../(tabs)/(home)/addOrEditCityWeather';
import WeatherIcons from '../(tabs)/(home)/chooseWeatherIcon';
import WeatherDetailsScreen from '../(tabs)/(home)/details';
import HomeScreen from '../(tabs)/(home)/index';
import migrations from '../../drizzle/migrations';

describe('Home screen', () => {
  beforeAll(async () => {
    await migrate(appDatabase, migrations);
  });

  afterAll(async () => {
    await wipeDatabase();
  });

  test('Cities renders correctly HomeScreen', async () => {
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue(cityFixture),
    });
    const render = renderRouter(
      {
        _layout: HomeLayout,
        details: WeatherDetailsScreen,
        index: HomeScreen,
        addOrEditCityWeather: AddOrEditCityWeather,
        chooseWeatherIcon: WeatherIcons,
      },
      {
        initialUrl: '/',
        wrapper: ({ children }) => (
          <Provider store={store}>
            <CitiesContextProvider>{children}</CitiesContextProvider>
          </Provider>
        ),
      },
    );

    await waitFor(() => screen.getAllByText('Vilnius'));
    expect(render.toJSON()).toMatchSnapshot();
  });
  test('Should display no data found when no cities are found', async () => {
    global.fetch = jest.fn().mockRejectedValue({
      json: jest.fn().mockRejectedValueOnce(new Error('Network error')),
    });

    await wipeDatabase();
    const render = renderRouter(
      {
        _layout: HomeLayout,
        details: WeatherDetailsScreen,
        index: HomeScreen,
        addOrEditCityWeather: AddOrEditCityWeather,
        chooseWeatherIcon: WeatherIcons,
      },
      {
        initialUrl: '/',
        wrapper: ({ children }) => (
          <Provider store={store}>
            <CitiesContextProvider>{children}</CitiesContextProvider>
          </Provider>
        ),
      },
    );
    await waitFor(() => screen.getAllByText('No data found'));

    const emptyState = screen.getByText('No data found');
    expect(emptyState).toBeOnTheScreen();
    expect(render.toJSON()).toMatchSnapshot();
  });
  test('Should navigate to new weather city screen when button is clicked ', async () => {
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue(cityFixture),
    });
    renderRouter(
      {
        _layout: HomeLayout,
        details: WeatherDetailsScreen,
        index: HomeScreen,
        addOrEditCityWeather: AddOrEditCityWeather,
        chooseWeatherIcon: WeatherIcons,
      },
      {
        initialUrl: '/',
        wrapper: ({ children }) => (
          <Provider store={store}>
            <PersistGate persistor={persistor}>
              <CitiesContextProvider>{children}</CitiesContextProvider>
            </PersistGate>
          </Provider>
        ),
      },
    );

    await waitFor(() => screen.getAllByText('Vilnius'));

    const addNewWeatherButton = screen.getByText('Add new');
    expect(addNewWeatherButton).toBeOnTheScreen();
    fireEvent(addNewWeatherButton, 'press');

    expect(screen).toHavePathname('/addOrEditCityWeather');
  });
});
