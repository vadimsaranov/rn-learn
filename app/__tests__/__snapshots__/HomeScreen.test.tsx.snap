// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen Cities renders correctly HomeScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        accessibilityElementsHidden={false}
        activityState={2}
        collapsable={false}
        fullScreenSwipeShadowEnabled={true}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        hasLargeHeader={false}
        importantForAccessibility="auto"
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onHeaderHeightChange={[Function]}
        onNativeDismissCancelled={[Function]}
        onSheetDetentChanged={[Function]}
        onTransitionProgress={[Function]}
        onWillAppear={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents={
          [
            1,
          ]
        }
        sheetCornerRadius={-1}
        sheetElevation={24}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetInitialDetent={0}
        sheetInitialDetentIndex={0}
        sheetLargestUndimmedDetent={-1}
        sheetLargestUndimmedDetentIndex={-1}
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": undefined,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenContentWrapper
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              [
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                {
                  "backgroundColor": "#fff",
                },
              ],
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <View>
              <RNGestureHandlerButton
                collapsable={false}
                enabled={true}
                hitSlop={
                  {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                  }
                }
                rippleColor={0}
                style={
                  [
                    {},
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "borderColor": "#b8b8b8",
                        "borderRadius": 12,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "gap": 4,
                        "justifyContent": "center",
                        "padding": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000",
                      },
                      undefined,
                    ]
                  }
                >
                  Add new
                </Text>
              </RNGestureHandlerButton>
            </View>
            <RCTScrollView
              ListEmptyComponent={[Function]}
              data={
                [
                  {
                    "cloudCover": 100,
                    "humidity": 80,
                    "icon": "04d",
                    "id": "1",
                    "isFavorite": false,
                    "name": "Vilnius",
                    "pressure": 1021,
                    "temp": 276.87,
                    "weather": {
                      "description": "overcast clouds",
                      "icon": "Clouds",
                      "id": undefined,
                      "main": undefined,
                    },
                    "weather_id": 1,
                    "wind": 3.6,
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onEndReached={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              onViewableItemsChanged={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={
                [
                  {
                    "onViewableItemsChanged": [Function],
                    "viewabilityConfig": undefined,
                  },
                ]
              }
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View>
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomColor": "#b8b8b8",
                          "borderBottomWidth": 1,
                          "flexDirection": "row",
                          "gap": 30,
                          "justifyContent": "space-between",
                          "paddingHorizontal": 16,
                          "paddingVertical": 8,
                        }
                      }
                    >
                      <Image
                        source={
                          {
                            "cache": "only-if-cached",
                            "uri": "https://openweathermap.org/img/wn/04d@2x.png",
                          }
                        }
                        style={
                          {
                            "height": 30,
                            "width": 30,
                          }
                        }
                      />
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              {
                                "color": "#000",
                                "fontSize": 17,
                              },
                            ]
                          }
                        >
                          Vilnius
                        </Text>
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              {
                                "color": "#8c8c8c",
                                "fontSize": 12,
                              },
                            ]
                          }
                        />
                      </View>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "gap": 12,
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "backgroundColor": "#90e0ef",
                              "borderRadius": 20,
                              "paddingHorizontal": 12,
                              "paddingVertical": 4,
                            }
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#000",
                                },
                                {
                                  "color": "#fff",
                                },
                              ]
                            }
                          >
                            276.87
                             °F
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
            <TrueSheetView
              background={4294967295}
              cornerRadius={24}
              dimmed={true}
              dismissible={true}
              edgeToEdge={false}
              grabber={true}
              initialIndexAnimated={true}
              keyboardMode="resize"
              onContainerSizeChange={[Function]}
              onDismiss={[Function]}
              onDragBegin={[Function]}
              onDragChange={[Function]}
              onDragEnd={[Function]}
              onMount={[Function]}
              onPresent={[Function]}
              onSizeChange={[Function]}
              scrollableHandle={null}
              sizes={
                [
                  "small",
                ]
              }
              style={
                {
                  "left": -9999,
                  "position": "absolute",
                  "width": "100%",
                  "zIndex": -9999,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  [
                    {
                      "height": undefined,
                      "overflow": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  collapsable={false}
                  onLayout={[Function]}
                >
                  <View
                    style={
                      {
                        "gap": 12,
                        "justifyContent": "center",
                        "paddingHorizontal": 16,
                        "paddingTop": 32,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#000",
                          },
                          {
                            "fontSize": 24,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Add city to favorites
                    </Text>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                        }
                      }
                    >
                      <RNGestureHandlerButton
                        collapsable={false}
                        enabled={true}
                        hitSlop={
                          {
                            "bottom": 0,
                            "left": 0,
                            "right": 0,
                            "top": 0,
                          }
                        }
                        rippleColor={0}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#b8b8b8",
                                "borderRadius": 12,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "gap": 4,
                                "justifyContent": "center",
                                "padding": 8,
                              },
                              false,
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              undefined,
                            ]
                          }
                        >
                          Add to favorites
                        </Text>
                      </RNGestureHandlerButton>
                      <RNGestureHandlerButton
                        collapsable={false}
                        enabled={false}
                        hitSlop={
                          {
                            "bottom": 0,
                            "left": 0,
                            "right": 0,
                            "top": 0,
                          }
                        }
                        rippleColor={0}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#b8b8b8",
                                "borderRadius": 12,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "gap": 4,
                                "justifyContent": "center",
                                "padding": 8,
                              },
                              {
                                "opacity": 0.4,
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              undefined,
                            ]
                          }
                        >
                          Remove from favorites
                        </Text>
                      </RNGestureHandlerButton>
                    </View>
                  </View>
                </View>
                <View
                  collapsable={false}
                  onLayout={[Function]}
                />
              </View>
            </TrueSheetView>
          </View>
        </RNSScreenContentWrapper>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleFontFamily="System"
          backTitleVisible={true}
          backgroundColor="#fff"
          color="#000"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleFontFamily="System"
          largeTitleFontWeight="700"
          largeTitleHideShadow={false}
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "absolute",
              "top": "-100%",
              "width": "100%",
            }
          }
          title="Weather"
          titleColor="#000"
          titleFontFamily="System"
          titleFontWeight="600"
          topInsetEnabled={false}
          translucent={false}
        />
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Home screen Should display no data found when no cities are found 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        accessibilityElementsHidden={false}
        activityState={2}
        collapsable={false}
        fullScreenSwipeShadowEnabled={true}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        hasLargeHeader={false}
        importantForAccessibility="auto"
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onHeaderHeightChange={[Function]}
        onNativeDismissCancelled={[Function]}
        onSheetDetentChanged={[Function]}
        onTransitionProgress={[Function]}
        onWillAppear={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents={
          [
            1,
          ]
        }
        sheetCornerRadius={-1}
        sheetElevation={24}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetInitialDetent={0}
        sheetInitialDetentIndex={0}
        sheetLargestUndimmedDetent={-1}
        sheetLargestUndimmedDetentIndex={-1}
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": undefined,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenContentWrapper
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              [
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                {
                  "backgroundColor": "#fff",
                },
              ],
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <View>
              <RNGestureHandlerButton
                collapsable={false}
                enabled={true}
                hitSlop={
                  {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                  }
                }
                rippleColor={0}
                style={
                  [
                    {},
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "borderColor": "#b8b8b8",
                        "borderRadius": 12,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "gap": 4,
                        "justifyContent": "center",
                        "padding": 8,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000",
                      },
                      undefined,
                    ]
                  }
                >
                  Add new
                </Text>
              </RNGestureHandlerButton>
            </View>
            <RCTScrollView
              ListEmptyComponent={[Function]}
              data={[]}
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onEndReached={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              onViewableItemsChanged={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={
                [
                  {
                    "onViewableItemsChanged": [Function],
                    "viewabilityConfig": undefined,
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#000",
                      },
                      undefined,
                    ]
                  }
                >
                  No data found
                </Text>
              </View>
            </RCTScrollView>
            <TrueSheetView
              background={4294967295}
              cornerRadius={24}
              dimmed={true}
              dismissible={true}
              edgeToEdge={false}
              grabber={true}
              initialIndexAnimated={true}
              keyboardMode="resize"
              onContainerSizeChange={[Function]}
              onDismiss={[Function]}
              onDragBegin={[Function]}
              onDragChange={[Function]}
              onDragEnd={[Function]}
              onMount={[Function]}
              onPresent={[Function]}
              onSizeChange={[Function]}
              scrollableHandle={null}
              sizes={
                [
                  "small",
                ]
              }
              style={
                {
                  "left": -9999,
                  "position": "absolute",
                  "width": "100%",
                  "zIndex": -9999,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  [
                    {
                      "height": undefined,
                      "overflow": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  collapsable={false}
                  onLayout={[Function]}
                >
                  <View
                    style={
                      {
                        "gap": 12,
                        "justifyContent": "center",
                        "paddingHorizontal": 16,
                        "paddingTop": 32,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#000",
                          },
                          {
                            "fontSize": 24,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Add city to favorites
                    </Text>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                        }
                      }
                    >
                      <RNGestureHandlerButton
                        collapsable={false}
                        enabled={true}
                        hitSlop={
                          {
                            "bottom": 0,
                            "left": 0,
                            "right": 0,
                            "top": 0,
                          }
                        }
                        rippleColor={0}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#b8b8b8",
                                "borderRadius": 12,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "gap": 4,
                                "justifyContent": "center",
                                "padding": 8,
                              },
                              false,
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              undefined,
                            ]
                          }
                        >
                          Add to favorites
                        </Text>
                      </RNGestureHandlerButton>
                      <RNGestureHandlerButton
                        collapsable={false}
                        enabled={false}
                        hitSlop={
                          {
                            "bottom": 0,
                            "left": 0,
                            "right": 0,
                            "top": 0,
                          }
                        }
                        rippleColor={0}
                        style={
                          [
                            {},
                            [
                              {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#b8b8b8",
                                "borderRadius": 12,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "gap": 4,
                                "justifyContent": "center",
                                "padding": 8,
                              },
                              {
                                "opacity": 0.4,
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              undefined,
                            ]
                          }
                        >
                          Remove from favorites
                        </Text>
                      </RNGestureHandlerButton>
                    </View>
                  </View>
                </View>
                <View
                  collapsable={false}
                  onLayout={[Function]}
                />
              </View>
            </TrueSheetView>
          </View>
        </RNSScreenContentWrapper>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleFontFamily="System"
          backTitleVisible={true}
          backgroundColor="#fff"
          color="#000"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleFontFamily="System"
          largeTitleFontWeight="700"
          largeTitleHideShadow={false}
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "absolute",
              "top": "-100%",
              "width": "100%",
            }
          }
          title="Weather"
          titleColor="#000"
          titleFontFamily="System"
          titleFontWeight="600"
          topInsetEnabled={false}
          translucent={false}
        />
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
